webapp/app/[locale]/tunnel/_hooks/useDeposit.ts

```diff
@@ -1,303 +1,221 @@
-    depositNativeTokenGasFees,
-    depositNativeTokenTxHash,
-    resetDepositNativeToken,
-  } = useDepositNativeToken({
-    enabled: depositingNative && canDeposit,
-    l1ChainId: fromToken.chainId,
-    onSuccess: onUserAcceptingDeposit,
-    toDeposit,
-  })
-
-  const {
-    approvalError,
-    approvalQueryKey,
-    approvalReceipt,
-    approvalReceiptError,
-    approvalTokenGasFees,
-    approvalTxHash,
-    depositErc20TokenError,
-    depositErc20TokenGasFees,
-    depositErc20TokenTxHash,
-    depositToken,
-    needsApproval,
-    resetApproval,
-    resetDepositToken,
-  } = useDepositToken({
-    amount: fromInput,
-    enabled: !depositingNative && canDeposit,
-    extendedApproval: extendedErc20Approval,
-    fromToken,
-    onApprovalSuccess,
-    onSuccess: onUserAcceptingDeposit,
-    toToken,
-  })
-
-  const {
-    data: depositReceipt,
-    error: depositReceiptError,
-    queryKey: depositQueryKey,
-  } = useWaitForTransactionReceipt({
-    hash: depositingNative ? depositNativeTokenTxHash : depositErc20TokenTxHash,
-  })
-
-  useReloadBalances({
-    fromToken,
-    status: depositReceipt?.status,
-  })
-
-  const clearDepositNativeState = useCallback(
-    function () {
-      // clear the deposit operation hash
-      resetDepositNativeToken()
-      // clear transaction receipt state
-      queryClient.removeQueries({ queryKey: depositQueryKey })
-    },
-    [depositQueryKey, queryClient, resetDepositNativeToken],
+  const { track } = useUmami()
+  const updateNativeBalanceAfterFees = useUpdateNativeBalanceAfterReceipt(
+    fromToken.chainId,
   )
+  const { l1WalletClient } = useL1WalletClient(fromToken.chainId)
 
-  const clearDepositTokenState = useCallback(
-    function () {
-      // clear the approval operation hash, if any
-      resetApproval?.()
-      // clear the deposit operation hash
-      resetDepositToken()
-      // clear approval receipt state
-      queryClient.removeQueries({ queryKey: approvalQueryKey })
-      // clear transaction receipt state
-      queryClient.removeQueries({ queryKey: depositQueryKey })
-    },
-    [
-      approvalQueryKey,
-      depositQueryKey,
-      queryClient,
-      resetApproval,
-      resetDepositToken,
-    ],
-  )
+  const depositingNative = isNativeAddress(fromToken.address)
 
-  useEffect(
-    function updateDepositStatusAfterFailure() {
-      if (!depositReceiptError) {
-        return
-      }
-      const deposit = deposits.find(
-        d =>
-          d.transactionHash === currentTxHash &&
-          d.status === EvmDepositStatus.DEPOSIT_TX_PENDING,
-      )
-      if (!deposit) {
-        return
-      }
-      clearDepositNativeState()
-      clearDepositTokenState()
+  const { allowanceQueryKey } = useNeedsApproval({
+    address: fromToken.address,
+    amount,
+    spender: l1StandardBridgeAddress,
+  })
 
-      updateDeposit(deposit, {
-        status: EvmDepositStatus.DEPOSIT_TX_FAILED,
+  return useMutation({
+    mutationFn() {
```

Comment:
What about using a name that is more descriptive?

Reply:
`mutationFn` is the name that `useMutation` expects 🤷🏽  can't do anything about it. However, I updated it so the inline function as a name [07b169d](https://github.com/hemilabs/ui-monorepo/pull/1125/commits/07b169d96299cc9f8d138d8808ce51c94d3a1c57)

Reply:
Oh! Didn't see it was an object property!